### PR TITLE
hpos-admin: log CAS mismatches, resolves #388

### LINKED ISSUE #388: Add additional logging to hpos-admin
Currently `hpos-admin` logs individual requests:

```
$ sudo ./result/bin/hpos-admin 
- - - [2020-02-21 17:48:13] "GET / HTTP/1.1" 404 341 0.017322
- - - [2020-02-21 17:48:23] "GET / HTTP/1.1" 404 341 0.000873
- - - [2020-02-21 17:48:24] "GET / HTTP/1.1" 404 341 0.001050
```

@peeech wants to add more logging. I wonder what exactly we should add? I guess we could print config value as it gets updated, but not sure if it makes sense to do so in production.

### PR DESCRIPTION
**Pull request by yegortimoshenko**
If we want to log anything else, let me know.

### CODE REVIEW

**Comment by alastairong** [comment]
Does it make sense to log when hpos-config.json is actually changed?

I experienced some issues previously where I could rename my HoloPort but later it would reset to the default.

**Comment by samrose** [comment]
The consistent test failure is replicated in this run on hydra https://hydra.holo.host/build/16672/nixlog/1

**Comment by samrose** [comment]
The log actually shows 

```
<3>machine# [    8.487367] hpos-admin[645]:   File "/nix/store/dbgk0zpnymw9zfzly8k91yrx9sjd64d1-hpos-admin.py", line 59
<3>machine# [    8.491381] hpos-admin[645]:     return '', 409
<3>machine# [    8.497529] hpos-admin[645]:          ^
<3>machine# [    8.502283] hpos-admin[645]: SyntaxError: invalid syntax
```

investigating

File: overlays/holo-nixpkgs/hpos-admin/hpos-admin.py [hunk 1]
```diff
@@ -54,7 +52,10 @@ def replace_file_contents(path, data):
 def put_settings():
     with state_lock:
         state = get_state_data()
-        if request.headers.get('x-hpos-admin-cas') != cas_hash(state['v1']['settings']):
+        expected_cas = cas_hash(state['v1']['settings'])
+        received_cas = request.headers.get('x-hpos-admin-cas')
+        if received_cas != expected_cas:
+            app.logger.warning('CAS mismatch: {} != {}'.format(received_cas, expected_cas)
```
Comment (samrose) [inline overlays/holo-nixpkgs/hpos-admin/hpos-admin.py]:
```suggestion
            app.logger.warning('CAS mismatch: {} != {}'.format(received_cas, expected_cas))
```

Comment (samrose) [reply overlays/holo-nixpkgs/hpos-admin/hpos-admin.py]: Tests pass with this change

Comment (samrose) [reply overlays/holo-nixpkgs/hpos-admin/hpos-admin.py]: @yegortimoshenko

**Comment by samrose** [comment]
Also 

```
flake8 ~/holo-nixpkgs/overlays/holo-nixpkgs/hpos-admin/hpos-admin.py
/home/samrose/holo-nixpkgs/overlays/holo-nixpkgs/hpos-admin/hpos-admin.py:58:80: E501 line too long (91 > 79 characters)
```

If you feel that is relevant, is the only issue after the closing paren issue above @yegortimoshenko

**Pull request merged**